lib/dialects/abstract/query.js

```diff
@@ -514,7 +514,19 @@ AbstractQuery.prototype.isUpdateQuery = function() {
 
 AbstractQuery.prototype.handleSelectQuery = function(results) {
   var result = null;
-
+  // Map raw fields to arbitrary names if a mapping is provided
+  if (this.options.fieldMap) {
```

Comment:
We could introduce a new option to toggle on or off the auto-mapping to model attributes behavior. Right now we assume we should map fields anytime the model option is set. 
But it would require users writing raw queries to set on the new option in addition to setting the model.